src/steps/user-groups/index.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import {
+  ACCOUNT_ENTITY_KEY,
+  Entities,
+  Relationships,
+  Steps,
+} from '../constants';
+import { createUserGroupEntity } from './converter';
+
+export async function fetchUserGroups({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateUserGroups(async (userGroup) => {
+    const userGroupEntity = createUserGroupEntity(userGroup);
+    if (userGroupEntity) {
+      await jobState.addEntity(userGroupEntity);
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.HAS,
+          from: (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity,
+          to: userGroupEntity,
+        }),
+      );
+    }
+  });
+}
+
+export async function buildUserGroupsAndUserRelationships({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(
+    { _type: Entities.USER_GROUP._type },
+    async (userGroupEntity) => {
+      const userGroupDetail = await apiClient.fetchUserGroupDetails(
+        userGroupEntity.id as string,
+      );
+
+      for (const user of userGroupDetail.users) {
+        const userEntity = await jobState.findEntity(
+          `${user.name}-${user.id.toString()}`,
+        );
```

Comment:
Can you encapsulate user_key value generation in function such as `getUserKey(...)` that can be exported from `/steps/user-groups/converters.ts` and then use it here?